src/NuGet.Core/NuGet.CommandLine.XPlat/ReportRenderers/Models/ListPackageProjectModel.cs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Evaluation;
+using NuGet.CommandLine.XPlat.Utility;
+using NuGet.ProjectModel;
+
+namespace NuGet.CommandLine.XPlat
+{
+    /// <summary>
+    /// Calculated project data model for list report
+    /// </summary>
+    internal class ListPackageProjectModel
+    {
+        private const string ProjectNameMSbuildProperty = "MSBuildProjectName";
+        internal List<ReportProblem> ProjectProblems { get; private set; }
+        internal string ProjectPath { get; private set; }
+        // Original packages
+        internal IEnumerable<FrameworkPackages> Packages { get; private set; }
+        // Calculated project model data for each targetframeworks
+        internal List<ListPackageReportFrameworkPackage> TargetFrameworkPackages { get; private set; }
+        internal string HttpSourceWarning { get; private set; }
+        internal string ProjectName { get; private set; }
+        internal Project Project { get; }
+        internal ListPackageReportModel ReportModel { get; }
+
+        public ListPackageProjectModel(string projectPath, ListPackageReportModel reportModel)
+        {
+            ProjectPath = projectPath;
+            ReportModel = reportModel;
+
+            //Open project to evaluate properties for the assets
+            //file and the name of the project
+            Project = MSBuildAPIUtility.GetProject(projectPath);
```

Comment:
If this is a model, then the computation probably shouldn't be happening here.

Reply:
Sure, addressed.